http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -324,29 +303,22 @@ object ClusterInitValues {
   val serviceAccountCredentialsFilename = "service-account-credentials.json"
   val customEnvVarFilename = "custom_env_vars.env"
 
-  def apply(googleProject: GoogleProject,
-            clusterName: ClusterName,
-            stagingBucketName: GcsBucketName,
+  def apply(cluster: Cluster,
             initBucketName: GcsBucketName,
-            clusterRequest: ClusterRequest,
+            serviceAccountKey: Option[ServiceAccountKey],
             dataprocConfig: DataprocConfig,
+            proxyConfig: ProxyConfig,
             clusterFilesConfig: ClusterFilesConfig,
             clusterResourcesConfig: ClusterResourcesConfig,
-            proxyConfig: ProxyConfig,
-            serviceAccountKey: Option[ServiceAccountKey],
-            userEmailLoginHint: WorkbenchEmail,
-            contentSecurityPolicy: String,
-            clusterImages: Set[ClusterImage],
-            stagingBucket: GcsBucketName,
-            welderEnabled: Boolean): ClusterInitValues =
+            contentSecurityPolicy: String): ClusterInitValues =
     ClusterInitValues(
-      googleProject.value,
-      clusterName.value,
-      stagingBucketName.value,
-      clusterImages.find(_.tool == Jupyter).map(_.dockerImage).getOrElse(""),
-      clusterImages.find(_.tool == RStudio).map(_.dockerImage).getOrElse(""),
+      cluster.googleProject.value,
+      cluster.clusterName.value,
+      cluster.dataprocInfo.map(_.stagingBucket.value).getOrElse(""),
```

Comment:
not necessarily this PR, but I feel like this function should return an `Either[String, ClusterInitValues]` and in the case where these required fields (such as stagingBucket) is `None` for some reason, we can error out